homeassistant/components/arcam_fmj/config_flow.py

```diff
@@ -16,12 +29,75 @@ class ArcamFmjFlowHandler(config_entries.ConfigFlow):
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
 
-    async def async_step_import(self, import_config):
-        """Import a config entry from configuration.yaml."""
-        entries = self.hass.config_entries.async_entries(DOMAIN)
-        import_key = _GETKEY(import_config)
-        for entry in entries:
-            if _GETKEY(entry.data) == import_key:
-                return self.async_abort(reason="already_setup")
+    async def _async_set_unique_id_and_update(self, host, port, uuid):
+        await self.async_set_unique_id(uuid)
+        self._abort_if_unique_id_configured({CONF_HOST: host, CONF_PORT: port})
+
+    async def _async_check_and_create(self, host, port):
+        client = Client(host, port)
+        try:
+            await client.start()
+        except ConnectionFailed:
+            raise AbortFlow("unable_to_connect")
```

Comment:
We prefer to `return self.async_abort(...)`. It leads to the same place in the end.